src/backend/functions/risk-edit.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of NER's PM Dashboard and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import middy from '@middy/core';
+import jsonBodyParser from '@middy/http-json-body-parser';
+import httpErrorHandler from '@middy/http-error-handler';
+import validator from '@middy/validator';
+import { Handler } from 'aws-lambda';
+import { PrismaClient, Role } from '@prisma/client';
+import {
+  buildNoAuthResponse,
+  buildNotFoundResponse,
+  buildSuccessResponse,
+  eventSchema,
+  riskEditInputSchemaBody
+} from 'utils';
+import { FromSchema } from 'json-schema-to-ts';
+
+const prisma = new PrismaClient();
+
+export const editRisk: Handler<FromSchema<typeof inputSchema>> = async ({ body }, _context) => {
+  const { userId, id, detail, resolved } = body;
+
+  // verify user is allowed to edit work packages
+  const user = await prisma.user.findUnique({ where: { userId } });
+  if (!user) return buildNotFoundResponse('user', `#${userId}`);
+  if (user.role === Role.GUEST) return buildNoAuthResponse();
+
+  // get the original risk and check if it exists
+  const originalRisk = await prisma.risk.findUnique({
+    where: { id }
+  });
+  if (originalRisk === null) {
+    return buildNotFoundResponse('Risk', `#${id}`);
+  }
+
+  // update the risk with the input fields
+  const updatedRisk = await prisma.risk.update({
+    where: {
+      id
+    },
+    data: {
+      detail,
+      isResolved: resolved
+    }
+  });
+
+  // return the updated risk
+  return buildSuccessResponse(updatedRisk);
```

Comment:
Maybe need transformer?

Reply:
Can you elaborate more on the transformer? Like I compared with other editing endpoints, and for project-edit endpoint, I saw that it also just returned buildsuccessresponse at the end.

Reply:
Yes, if I recall correctly, the response from those endpoints isn't used by any of the front-end, so it doesn't matter how the response object is formed. But if we want to do it properly and use the response object, we should ensure that it is properly formatted. 

For the get endpoints, you should be able to observe that a transformer function, typically with several helper functions, is used to define the proper output object. 

One such example is that the database object may include a WBS_Element ID, but this is typically not something we want to provide to the front-end, so in the transformer function, we do not include it as a field in the return object. 

Another good example is WBS number. In the database, it is stored as 3 fields with a combined unique constraint. In the transformer functions, we typically have a helper function that takes in a WBS_Element database object and returns a proper WBS Number object for the front-end to use. Notably, the front-end expected WBS number to be an object containing the 3 fields, rather than just the 3 fields.